tools/filetools.go

```diff
@@ -121,3 +126,250 @@ func VerifyFileHash(oid, path string) error {
 
 	return nil
 }
+
+// FilenamePassesIncludeExcludeFilter returns whether a given filename passes the include / exclude path filters
+// Only paths that are in includePaths and outside excludePaths are passed
+// If includePaths is empty that filter always passes and the same with excludePaths
+// Both path lists support wildcard matches
+func FilenamePassesIncludeExcludeFilter(filename string, includePaths, excludePaths []string) bool {
+	if len(includePaths) == 0 && len(excludePaths) == 0 {
+		return true
+	}
+
+	if len(includePaths) > 0 {
+		matched := false
```

Comment:
But that conditional guards a lot more than just setting `matched := false`, in fact `matched` is deliberate scoped inside that block. It's that way because early-out by lack of inclusion requires checking every item and only concluding at the end, as opposed to early-out by exclusion which can return as soon as it's matched. That's why the exclusion block is simpler.